src/inmanta/graphql/schema.py

```diff
@@ -101,14 +61,60 @@ def add_filter_and_sort(
     return stmt
 
 
-def initialize_schema() -> strawberry.Schema:
+def get_schema(metadata: GraphQLMetadata) -> strawberry.Schema:
     """
     Initializes the Strawberry GraphQL schema.
     It is initiated in a function instead of being declared at the module level, because we have to do this
     after the SQLAlchemy engine is initialized.
     """
+
+    mapper: StrawberrySQLAlchemyMapper[typing.Any] = StrawberrySQLAlchemyMapper()
     loader = StrawberrySQLAlchemyLoader(async_bind_factory=get_session_factory())
 
+    def get_expert_mode(root: "Environment") -> bool:
+        """
+        Checks settings of environment to figure out if expert mode is enabled or not
+        """
+        assert hasattr(root, "settings")  # Make mypy happy
+        return bool(root.settings.get("enable_lsm_expert_mode", False))
```

Comment:
This I don't understand. What is the return type of `root.settings.get()`.

* If it's a boolean, the type cast is redundant.
* If it's a string, the type cast is incorrect.

I might be missing something here.

Reply:
`root.settings` is a `dict[str, Union[bool, int, float, str, dict[str, Union[str, int, bool]]]]`. I believe for this setting in particular it should be a bool. I'll make the comparison more robust either way